### ISSUE TITLE
need mechanism to reload value with current args

### ISSUE THREAD

**Issue by ehaynes99** [slorber/react-async-hook#18]
In the docs, there is this example for refresh/refetch data:
```typescript
const StarwarsHero = ({ id }) => {
  const asyncHero = useAsync(fetchStarwarsHero, [id]);

  return <div onClick={() => asyncHero.execute()}>...</div>;
};
```

However, as this is written, `id` would be undefined, as execute requires the arguments. It would actually have to be called as `asyncHero.execute(...asyncHero.currentParams)`. Could an actual `refresh` function be added that does this?

**Comment by slorber** [comment]
Hi,

Yes a refresh function could make sense. 

I need a variant that does not use params because "useAsyncCallback" is always executed manually and there's no concept of "current params" because you don't pass any params array for it. But there can be 2 methods for useAsync and only 1 for useAsyncCallback

**Comment by ehaynes99** [comment]
True. In general, i would consider `useAsync` to be akin to `useMemo`. Conceptually, a memo is always an expression, vs a callback which is a statement that might be an expression. In simpler terms, a memo IS something, and a callback DOES something. The concept of "reload" would only apply to the former. The latter might not even return a value.

**Comment by slorber** [comment]
yes useAsync is somehow useMemo for derived remote data :)
